src/io/reader.rs

```diff
@@ -0,0 +1,108 @@
+use std::fs::File;
+use std::io::{self, BufRead, BufReader, Cursor, Read, Seek, SeekFrom};
+use std::path::Path;
+
+use dynimage::DynamicImage;
+use image::ImageFormat;
+use {ImageError, ImageResult};
+
+use super::free_functions;
+
+/// A multi-format image reader.
+///
+/// Wraps an input reader to facilitate automatic detection of an image's format, appropriate
+/// decoding method, and dispatches into the set of supported `ImageDecoder` implementations.
+pub struct Reader<R: Read> {
```

Comment:
I prefer something like `ImageReader` just to give users a bit more context.

Reply:
The type is available from a module with full path `image::io::Reader`. That should be reasonable context, it can be imported `as ImageReader` if the user requires even more. Overall, it seems to be similar to [RFC356](https://github.com/aturon/rfcs/blob/no-module-prefixes/active/0000-no-module-prefixes.md) on naming conventions.